dedupe/record_linkage_testing/Dockerfile.api

```diff
@@ -0,0 +1,16 @@
+ARG VERSION=v1.2.11
+# The record linkage image is not specifically built for the arm64 architecture,
+# so we need to explicitly state the amd64 architecture here so users running
+# Docker on Apple Silicon know to build the image for the correct architecture.
+FROM --platform=linux/amd64 ghcr.io/cdcgov/phdi/record-linkage:${VERSION}
```

Comment:
When running on Apple Silicon, I got the following error:
```
ERROR [api internal] load metadata for ghcr.io/cdcgov/phdi/record-linkage:v1.2.11:
failed to solve: ghcr.io/cdcgov/phdi/record-linkage:v1.2.11: no match for platform in manifest: not found
```

After pairing with @ericbuckley, specifying the platform explicitly by adding `--platform=linux/amd64` to this line allowed tests to succeed.